staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go

```diff
@@ -104,23 +103,11 @@ func logsForObjectWithClient(clientset corev1client.CoreV1Interface, object, opt
 			}
 
 			if currOpts.Container == "" {
-				// We don't know container name. In this case we expect only one container to be present in the pod (ignoring InitContainers).
```

Comment:
Agreed that we should refactor but we can save that until the old annotation is removed in 1.25. We should also eventually have these functions take in buffers to write to instead of grabbing `os.Stderr`.